### ISSUE TITLE
Add normalization options

### ISSUE THREAD

**Issue by vgarciasc** [vgarciasc/hundred-hammers#7]
It is very common to normalize data before training any models. However it is necessary to do this carefully, using the data in the training set to normalize the test set. Since HH abstracts away the whole training/testing separation process, it is not possible for the user to normalize data correctly given the current state of the library. It would be interesting to allow the user to determine something similar to a [sklearn Pipeline](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html).

**Comment by eugenioLR** [comment]
I think using an sklearn pipeline inside the code would make the hyperparameter optimization a bit to difficult to manage, since you can't get the parameters from the model directly. However, we can return the models as a pipeline.

I'm gonna get to it.
